go/worker/registration/worker.go

```diff
@@ -1094,6 +1094,18 @@ func New(
 		rp.SetAvailable(func(*node.Node) error { return nil })
 	}
 
+	// When a node is a client node and it has an entity configured, we register it with the
+	// observer role as this may be needed for confidential runtimes.
```

Comment:
Can you provide more details about the observer role?

Currently, the only change I see is in `ComputeRuntimeAllowedRoles`, which permits observers to register compute runtimes. If observers are passive compute nodes, they should likely also participate in P2P topics.

Should the observer role and the compute role be mutually exclusive?

Reply:
> If observers are passive compute nodes, they should likely also participate in P2P topics.

They are basically "client nodes that are registered on chain so that their eligibility can be enforced by an on-chain policy". They already participate in some P2P topics (e.g. the transaction propagation ones).

> Should the observer role and the compute role be mutually exclusive?

Yes, at least it doesn't make sense for a node to register for both (which should already be the case in the code above). Or do you think that a node should not even be allowed to register for both? I guess it doesn't make much sense to do so.

Reply:
I was thinking the node should not be allowed to register for both. But I guess this should not be a problem.